### ISSUE TITLE
no-unused-variable rule ignores decorated members

### ISSUE THREAD

**Issue by igabesz** [palantir/tslint#4013]
This is a feature request, a non-breaking extension to the current behavior. 

**Problem:** Decorators often change the behavior and usage of the decorated member. I have examples where some of my private member functions are called through decorators only -- so they appear to be unused but they are called indirectly through the decorators. 

```typescript
class CatchySong {
  @callMeMaybe
  private heresMyNumber() { throw new Error('meh'); } // tslint:disable-line:no-unused-variable
}
```

**Current Workaround:** Now I must make them public to avoid linter errors which is undesirable. Or I can disable tslint every such place (see above) which is pretty inconvenient.

**Feature Request:** I have read #667 (similar request but not so specific) and I agree that the default operation of no-unused-variable rule should NOT be changed. However, the rule could be extended by an `ignore-decorated` option like the currently available `ignore-pattern`. If `ignore-decorated` is set to `true` then decorated members are always accepted as valid whether they are used or not. This option is turned off by default. 

Config example:

```json
{
  "rules": {
    "no-unused-variable": {
        "severity": "warning",
        "options": {"ignore-pattern": "^_", "ignore-decorated": true }
    },
  }
}
```

**Comment by igabesz** [comment]
I see no such feature or issue in eslint:
https://github.com/eslint/eslint/issues?q=is%3Aissue+no-unused-vars+decorators

**Comment by igabesz** [comment]
A better workaround is if such members are `protected` instead of being `public`. This way they are hidden from outside but it's still fishy because they are not intended to be used through inheritance -- probably there is no inheritance at all for the given class. +1 for better code-completion, -1 for worse clarity.

**Comment by johnwiseheart** [comment]
`no-unused-variable` has been deprecated, since typescript can handle its functionality now.
